src/test/java/com/agnieszkapawska/flashcards/domain/services/FlashcardServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package com.agnieszkapawska.flashcards.domain.services;
+
+import com.agnieszkapawska.flashcards.FlashcardAndQuestionTagAbstractTests;
+import com.agnieszkapawska.flashcards.domain.models.Flashcard;
+import com.agnieszkapawska.flashcards.domain.repositories.FlashcardRepository;
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import javax.persistence.EntityNotFoundException;
+import java.util.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class FlashcardServiceTest extends FlashcardAndQuestionTagAbstractTests {
+    @MockBean
+    private FlashcardRepository flashcardRepository;
+    @Autowired
+    private FlashcardService flashcardService;
+
+    @Test
+    public void shouldInvokeMethodOnFlashcardRepository_WhenSaveFlashcard() {
+        //given
+
+        //when
+        flashcardService.saveFlashcard(new Flashcard());
```

Comment:
You could extract Flashcard to variable and then use it in assert to be sure that repository is called with exactly that instance.